specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/stable/2024-06-30/dedicatedhsm.json

```diff
@@ -0,0 +1,762 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Get information about all dedicated HSMs in the specified resource group.",
+            "schema": {
+              "$ref": "#/definitions/DedicatedHsmListResult"
+            }
+          },
+          "default": {
+            "description": "The error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/DedicatedHsmError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List dedicated HSM devices in a resource group": {
+            "$ref": "./examples/DedicatedHsm_ListByResourceGroup.json"
+          },
+          "List dedicated HSM devices in a resource group including payment HSM": {
+            "$ref": "./examples/PaymentHsm_ListByResourceGroup.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.HardwareSecurityModules/dedicatedHSMs": {
+      "get": {
+        "tags": [
+          "DedicatedHsms"
+        ],
+        "operationId": "DedicatedHsm_ListBySubscription",
+        "description": "The List operation gets information about the dedicated HSMs associated with the subscription.",
+        "parameters": [
+          {
+            "name": "$top",
+            "in": "query",
+            "type": "integer",
+            "format": "int32",
+            "description": "Maximum number of results to return."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Get information about all dedicated HSMs in the specified subscription.",
+            "schema": {
+              "$ref": "#/definitions/DedicatedHsmListResult"
+            }
+          },
+          "default": {
+            "description": "The error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/DedicatedHsmError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List dedicated HSM devices in a subscription": {
+            "$ref": "./examples/DedicatedHsm_ListBySubscription.json"
+          },
+          "List dedicated HSM devices in a subscription including payment HSM": {
+            "$ref": "./examples/PaymentHsm_ListBySubscription.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HardwareSecurityModules/dedicatedHSMs/{name}/outboundNetworkDependenciesEndpoints": {
```

Comment:
>/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HardwareSecurityModules/dedicatedHSMs/{name}/outboundNetworkDependenciesEndpoints [](http://example.com/codeflow?start=5&length=168)

This resource type is missing point GET, PUT and DELETE.

Reply:
This API returns the network endpoints of all outbound dependencies in the specified dedicated hsm resource. PUT and DELETE operations are not needed. documentation [link](https://learn.microsoft.com/en-us/cli/azure/dedicated-hsm?view=azure-cli-latest#az-dedicated-hsm-list-outbound-network-dependency-endpoint)